docs/proposals/configmap-rollout.md

```diff
@@ -0,0 +1,171 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# ConfigMap management
+
+## Abstract
+
+A proposal for implementing management of ConfigMaps in the context of a
+[Deployment](http://kubernetes.io/docs/user-guide/deployments/).
+
+Users should be able to have their apps automatically pick up changes to
+their configuration and also have the ability to rollback to previous
+configuration similar to the ability to rollback to previous pod template
+revisions.
+
+Currently, in order to apply updated configuration into a running app, users
+need to 1) either do it manually: copy the ConfigMap that holds their old
+behavior, update it, store it back as a different ConfigMap, and mount the
+new ConfigMap into the Deployment, or 2) edit the original ConfigMap and
+trigger the deployment controller to start a new ReplicaSet (either by
+adding a dummy environment value in the pod template of the Deployment or
+by including the update as part of another update) losing the ability to
```

Comment:
if they change the config map name, they have to update the deployment pod template spec, which will trigger the deployment.